### ISSUE TITLE
osh hangs after nano and sometimes other outputs

### ISSUE THREAD

**Issue by dezren39** [oils-for-unix/oils#2040]
![image](https://github.com/user-attachments/assets/1adc2aaa-6e57-4401-a0b4-d86b71662688)
after ctrl+x,y it freezes like this.

also 
```
usage() {
    echo "Usage: $0 <source_directory> <target_directory>"
    echo "Moves files from source_directory to target_directory, handling duplicates by appending _1, _2, etc."
    exit 1
}
```
i had a script that runs the above and when it exits like this it also hung up.

**Comment by dezren39** [comment]
it's almost like, if i run a command with multiple lines of output, then do something else, the terminal cursor gets stuck on a previously printed line and then won't let me type or enter things, keys entered look kind of like they are deleted or pre-empted

**Comment by andychu** [comment]
hm

- what is ctrl-x, y?   do you have some custom key bindnigs?
- do you have anything in your oshrc?
  - try with `osh --rcfile /dev/null` .  Does the problem still appear
- what is `osh --version`?
  - Are you using the Python or C++ version?  We are moving to C++, but unfortunately Python is packaged in more places

**Comment by dezren39** [comment]
```
osh-0.22.0$ osh --version
Oils 0.22.0		https://www.oilshell.org/

git commit = 19ddf310bd79f5b0a62b57eee9c41209474e2888
```
i think the cpp version? from nixpkgs/master.

i don't think i have any conflicting keys, definitely not from hyprland but i'll double check kitty.

i don't see it in my short test just now without an rc file, but i dont think i set an rc file. it doesn't happen every time though so i'll need to try for a little while before i know it doesn't happen.

**Comment by McPherran** [comment]
Yes, after exiting/suspending nano I have seen hang on occasion and also (more commonly) the command line text will become messy. E.g. text missing/overwritten, wrong indentation, etc. **A helpful tip:** Ctrl+L (lowercase 'l') clears the terminal/console and may clear issues that are encountered.

**Comment by andychu** [comment]
Hm that should not happen

If you can post steps for repro here, with a screenshot, that might help us recognize the bugs

- #2555 

Or at least if others encountered similar issues, they can tell if it's the same bug

**Comment by McPherran** [comment]
It happens with nano and not so much (if at all) with vim. It's fairly simple to reproduce. I opened a terminal emulator window, then ran nano and exited. Then make the window larger. Run nano and right off the bat one can see that nano uses the smaller (original) window size. Exit nano in you may see artifacts (title bar, help bar) from nano remaining in the console. Combine some of these steps with ls -Al of a long directory and you may be able to see that a directory output remains on the cursor line and becomes convoluted with a typed command (^L will clear window to return to a better state).

**$LINES and $COLUMNS are not updated when the window is resized.** This is probably the root/major cause.
